packages/datastore/amplify_datastore_plugin_interface/test/amplify_modelschema_test.dart

```diff
@@ -71,7 +71,7 @@ void main() {
         association: ModelAssociation(
           associationType: ModelAssociationType.hasMany,
           associatedName: Post.BLOG.fieldName,
-          associatedType: Post.BLOG.fieldType!.ofModelName,
+          associatedType: 'Post',
```

Comment:
ModelField.== didn't use to compare associations, that's why these values were wrong/missing
https://github.com/aws-amplify/amplify-flutter/blob/58660b1b1c77ae1b8773a7b29bc8c30312369e65/packages/amplify_core/lib/src/types/models/model_field.dart#L98